runtime/src/telemetry/metrics/task.rs

```diff
@@ -21,27 +19,24 @@ impl Label {
         Self {
             name: String::new(),
             kind: Kind::Root,
-            supervision: Supervision::Detached,
-            mode: Mode::Shared,
+            execution: Execution::Shared,
         }
     }
 
     /// Create a new label for a future task.
-    pub fn task(name: String, supervised: bool, dedicated: bool, blocking: bool) -> Self {
+    pub fn task(name: String, execution: crate::Execution) -> Self {
         Self {
             name,
             kind: Kind::Task,
-            supervision: if supervised {
-                Supervision::Supervised
-            } else {
-                Supervision::Detached
-            },
-            mode: if dedicated {
-                Mode::Dedicated
-            } else if blocking {
-                Mode::SharedBlocking
-            } else {
-                Mode::Shared
+            execution: match execution {
+                crate::Execution::Dedicated => Execution::Dedicated,
+                crate::Execution::Shared(blocking) => {
```

Comment:
Small nit - we're using `crate::Execution` in a few places, let's just import it.